src/count-dependency-imports.ts

```diff
@@ -0,0 +1,96 @@
+import fs from "fs";
+import path from "path";
+import { glob } from "glob";
+import { $ } from "bun";
+
+export async function countDependencyImports(dir: string) {
+  if (!dir) {
+    console.error("Please provide a directory path as argument.");
+    process.exit(1);
+  }
+
+  // read the deps.json from dir.
+  // if it doesn't exist, create it.
+  if (!fs.existsSync(path.join(dir, "deps.json"))) {
+    console.log(
+      "deps.json not found in the current directory, creating it with nx"
+    );
+    await $`cd ${path.join(dir)} && npx nx graph --file=deps.json`;
+  }
+
+  console.log("Reading deps.json");
+  const deps = JSON.parse(fs.readFileSync(path.join(dir, "deps.json"), "utf8"));
+
+  const packageDeps = Object.keys(deps.graph.dependencies);
+
+  console.log(`Analyzing ${packageDeps.length} dependencies...`);
+
+  packageDeps.forEach((dep) => {
+    const dependencies = deps.graph.dependencies[dep].map(
+      (d: { target: string }) => d.target
+    );
+    console.log(
+      `Analyzing ${dep}, contains ${dependencies.length} dependencies total, counting imports for each...`
+    );
+
+    if (dependencies.length === 0) {
+      console.log("No dependencies, skipping.");
+      return;
+    }
+    // for each dependency in deps, count the number of times it is imported.
+    countDependencies(dep, dependencies);
+    console.log("\n");
+  });
+
+  function countDependencies(pkg: string, dependencies: any[]) {
+    const packageDir = path.join(dir, "packages", pkg);
+    let actualDir = packageDir.replace("@glide/", "");
+
+    if (pkg === "@glide/app") {
+      actualDir = path.join(dir, "app");
+    }
+
+    if (pkg === "@glide/functions") {
+      actualDir = path.join(dir, "functions");
+    }
+
+    const tsFiles = glob.sync(path.join(actualDir, "**/*.ts*"));
+
+    // Use a Map to store and count dependencies
+    const dependencyCounts = new Map();
+
+    // Updated regex to capture the full import path
+    const importExportRegex = /from\s+['"](@?[^'"]+)['"]/g;
```

Comment:
I recommend using `parseImports`

Reply:
We also have `walkDirectory`, which in hindsight should actually just use globbing, not walking.